flowtorch/bijectors/base.py

```diff
@@ -215,3 +222,54 @@ def inverse_shape(self, shape: torch.Size) -> torch.Size:
         Defaults to preserving shape.
         """
         return shape
+
+    def invert(self) -> Bijector:
+        return InverseBijector(self)
+
+
+class InverseBijector(Bijector):
+    """
+    An inverse bijector class.
+    InverseBijector flips a bijector such that forward calls inverse and
+    inverse calls forward.
+    The log-abs-det-Jacobian is updated accordingly.
+
+    Args:
+        bijector (Bijector): layer to be inverted
+
+    Examples:
+
+    """
+
+    def __init__(self, bijector: Bijector) -> None:
```

Comment:
What do you think about implementing `InverseBijector` so it can be applied to a lazy bijector analogous to how `Compose` works?

At the moment, I think this requires a concrete bijector so you can't do e.g.:
```
bijectors = B.Inverse(B.AffineAutoregressive(params_fn=params_fn))
```

Reply:
I reckon it would be more useful to have an `Inverse` class that can invert your flow plan, so you can e.g. turn Inverse Autoregressive Flow (IAF) into Masked Autoregressive Flow (MAF) rather than swap the operators after the bijector's been created.I think you rarely deal with a concrete bijector object, it will be taken care of by `flowtorch.distributions.Flow`. What do you think?

Reply:
I was imagining it would go like this:
```
class Invert(Bijector):
    def __init__(
        self,
        bijector: flowtorch.Lazy,
        *,
        shape: torch.Size,
        context_shape: Optional[torch.Size] = None,
    ):
```